modules/core/src/test/java/org/apache/ignite/internal/processors/security/sandbox/IgnitionComponentProxyTest.java

```diff
@@ -0,0 +1,191 @@
+                return Ignition.localIgnite();
+            }
+        };
+
+        check(s);
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void testGetOrStart() {
+        Supplier<Ignite> s = new Supplier<Ignite>() {
+            @Override public Ignite get() {
+                try {
+                    return Ignition.getOrStart(Ignition.localIgnite().configuration());
+                }
+                catch (Exception e) {
+                    throw new IgniteException(e);
+                }
+            }
+        };
+
+        check(s);
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void testIgniteWithName() {
+        Supplier<Ignite> s = new Supplier<Ignite>() {
+            @Override public Ignite get() {
+                return Ignition.ignite(SRV);
+            }
+        };
+
+        check(s);
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void testIgniteWithID() {
+        final UUID srvId = grid(SRV).localNode().id();
+
+        Supplier<Ignite> s = new Supplier<Ignite>() {
+            @Override public Ignite get() {
+                return Ignition.ignite(srvId);
+            }
+        };
+
+        check(s);
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void testAllGrids() {
+        Supplier<Collection<Ignite>> s = new Supplier<Collection<Ignite>>() {
+            @Override public Collection<Ignite> get() {
+                return Ignition.allGrids();
+            }
+        };
+
+        checkCollection(s);
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void testStart(){
+        Supplier<Ignite> s = new Supplier<Ignite>() {
+            @Override public Ignite get() {
+                try {
+                    return Ignition.start(getConfiguration("node_" + G.allGrids().size()));
+                }
+                catch (Exception e) {
+                    throw new IgniteException(e);
+                }
+            }
+        };
+
+        check(s);
+    }
+
+    /**
+     * @param s Supplier that returns an instance of Ignite.
+     */
+    private void check(final Supplier<Ignite> s) {
+        Supplier<Collection<Ignite>> supplier = new Supplier<Collection<Ignite>>() {
+            @Override public Collection<Ignite> get() {
+                return Collections.singleton(s.get());
+            }
+        };
+
+        checkCollection(supplier);
+    }
+
+    /**
+     * @param s Supplier that returns a collection of Ignite instances.
+     */
+    private void checkCollection(Supplier<Collection<Ignite>> s) {
+        Ignite srv = grid(SRV);
+
+        Ignite clnt = grid(CLNT);
+        //Checks that inside the sandbox we should get a proxied instance of Ignite.
+        clnt.compute(clnt.cluster().forNodeId(srv.cluster().localNode().id()))
+            .broadcast(() -> {
+                Collection<Ignite> nodes = s.get();
+
+                for (Ignite node : nodes) {
+                    assertTrue(Proxy.isProxyClass(node.getClass()));
```

Comment:
Proxy it's not a goal itself. We should check that we can perform some operations using these ignite instances which we can't perform without proxy.